code/obj/item/organs/tail.dm

```diff
@@ -33,6 +33,14 @@
 			build_mob_tail_image()
 			update_tail_icon()
 
+	disposing()
+		on_removal()
```

Comment:
Should this be called always? What if the tail is not attached to anyone?

Reply:
I've restructured the code a little to compensate